File: framework/languages/java/samples/java/TicTacToe.SessionGateway/src/main/java/systems/zlink/samples/tictactoe/sessiongateway/server/play/gamespots/handlers/TicTacToeGameSpotActorLeftHandler.java
package systems.zlink.samples.tictactoe.sessiongateway.server.play.gamespots.handlers;

import systems.zlink.samples.tictactoe.sessiongateway.server.play.gamespots.TicTacToeMatchRoom;

public final class TicTacToeGameSpotActorLeftHandler {
    public TicTacToeMatchRoom handle(TicTacToeMatchRoom room, String actorId) {
        return new TicTacToeMatchRoom(room.matchId(), room.lastActorId().equals(actorId) ? "" : room.lastActorId());
    }
}

File: framework/languages/java/samples/kotlin/TicTacToe.SessionGateway/Server/Play/src/main/kotlin/systems/zlink/samples/kotlin/tictactoe/sessiongateway/server/play/gamespots/handlers/TicTacToeGameSpotActorLeftHandler.kt
package systems.zlink.samples.kotlin.tictactoe.sessiongateway.server.play.gamespots.handlers

import systems.zlink.samples.kotlin.tictactoe.sessiongateway.server.play.gamespots.TicTacToeMatchRoom

class TicTacToeGameSpotActorLeftHandler {
    fun handle(room: TicTacToeMatchRoom, actorId: String): TicTacToeMatchRoom =
        if (room.lastActorId == actorId) room.copy(lastActorId = "") else room
}
